libkrun/libkruntime.c

```diff
@@ -29,15 +29,70 @@
 #define ACTUAL_CLOCK_MONOTONIC    CLOCK_MONOTONIC
 #endif
 
+#define CORE_BOUNDS_CHECK(C)                                                \
```

Comment:
The compiler will almost certainly inline them, but I agree, the tiny performance difference either way is completely irrelevant -- let's favour readability.